### PR TITLE
Logging: Redirect StdErr into logging system

### PR DESCRIPTION
**Pull request by merryhime**
Redirect stderr output into logging system.

This allows for easier mvk/mesa debugging since stderr will now show up in log files.

Extracted from #4227.

### CODE REVIEW

**Review by marysaka — CHANGES_REQUESTED** [review]
I would prefer not importing Mono.Unix again as this force us to have another native dependency again :sweat_smile:

**Comment by merryhime** [comment]
@marysaka removed dependency on Mono.Unix

**Review by marysaka — APPROVED** [review]

File: Ryujinx.Common/SystemInterop/StdErrAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        [LibraryImport("libc", SetLastError = true)]
+        private static partial int dup2 (int fd, int fd2);
```
Comment (riperiperi) [inline Ryujinx.Common/SystemInterop/StdErrAdapter.cs:71]:
```suggestion
        private static partial int dup2(int fd, int fd2);
```

File: Ryujinx.Common/SystemInterop/StdErrAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            Span<int> pipefd = stackalloc int[2];
+            fixed (int* ptr = pipefd)
+            {
+                if (pipe(ptr) == 0)
+                {
+                    return (pipefd[0], pipefd[1]);
+                }
+                else
+                {
+                    throw new ();
```
Comment (riperiperi) [inline Ryujinx.Common/SystemInterop/StdErrAdapter.cs:87]:
```suggestion
                    throw new();
```

File: Ryujinx.Common/SystemInterop/UnixStream.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Ryujinx.Common.SystemInterop
+{
+    public partial class UnixStream : Stream, IDisposable
```
Comment (TSRBerry) [inline Ryujinx.Common/SystemInterop/UnixStream.cs:7]:
```suggestion
    [UnsupportedOSPlatform("windows")]
    public partial class UnixStream : Stream, IDisposable
```
Just to make sure we never use it for Windows.

File: Ryujinx.Common/SystemInterop/StdErrAdapter.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        public StdErrAdapter()
+        {
+            if (OperatingSystem.IsLinux() || OperatingSystem.IsMacOS())
+            {
+                RegisterPosix();
+            }
+        }
+
+        [SupportedOSPlatform("linux")]
+        [SupportedOSPlatform("macos")]
```
Comment (TSRBerry) [inline Ryujinx.Common/SystemInterop/StdErrAdapter.cs:26]: Could you move these to the class?

**Review by TSRBerry — COMMENTED** [review]
Just have a few suggestions, but lgtm otherwise!

File: Ryujinx.Common/SystemInterop/UnixStream.cs (hunk 3 above)
Comment (merryhime) [reply Ryujinx.Common/SystemInterop/UnixStream.cs:7]: sgtm

File: Ryujinx.Common/SystemInterop/StdErrAdapter.cs (hunk 4 above)
Comment (merryhime) [reply Ryujinx.Common/SystemInterop/StdErrAdapter.cs:26]: Would rather leave this be, and have this check be encapsulated in this class.

Comment (TSRBerry) [reply Ryujinx.Common/SystemInterop/StdErrAdapter.cs:26]: Fine for me as well!

**Review by AcK77 — APPROVED** [review]

**Review by TSRBerry — APPROVED** [review]
Thank you!

**Pull request merged**